cmd/cluster/core/dump.go

```diff
@@ -262,7 +226,7 @@ func createGuestKubeconfig(ctx context.Context, c client.Client, cpNamespace str
 	}
 
 	for k := range localhostKubeconfig.Clusters {
-		localhostKubeconfig.Clusters[k].Server = fmt.Sprintf("https://localhost:%d", localPort)
+		localhostKubeconfig.Clusters[k].Server = fmt.Sprintf("https://kube-apiserver.%s.svc.cluster.local:6443", cpNamespace)
```

Comment:
Fwiw this was introduced here https://github.com/openshift/hypershift/pull/2571 to support private clusters by @csrwng 

The assumption that the svc network is reachable is not always true. I think we might want to make this a flag e.g. use-service-network